tests/production_pause_vote_timeout.py

```diff
@@ -0,0 +1,200 @@
+
+    Print("Restart finalizercNode")
+    finalizercNode.relaunch()
+
+    Print("Verify LIB advances after restart of finalizercNode")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert node1.waitForLibToAdvance(), "node1 did not advance LIB"
+    assert producercNode.waitForLibToAdvance(), "producercNode did not advance LIB"
+
+    ####################### test 2 ######################
+
+    Print("Shutdown centerNode")
+    centerNode.kill(signal.SIGTERM)
+    assert not centerNode.verifyAlive(), "centerNode did not shutdown"
+
+    # wait some time for producercNode paused
+    paused = False
+    for i in range(0, 15):
+        time.sleep(1)
+        paused = producercNode.processUrllibRequest("producer", "paused", returnType=ReturnType.raw)
+        if paused == b'true':
+            Print(f'paused after {i} seconds after centerNode was shutdown')
+            break;
+    # Verify producercNode paused
+    assert paused, "producercNode still producing after centerNode was shutdown"
+    # Verify node0 and node1 still producing but LIB should not advance
+    assert node0.processUrllibRequest("producer", "paused", returnType=ReturnType.raw) == b'false', "node0 paused after centerNode was shutdown"
+    assert node1.processUrllibRequest("producer", "paused", returnType=ReturnType.raw) == b'false', "node1 paused after centerNode was shutdown"
+    if node0.waitForLibToAdvance(timeout=5): # LIB can advance for a few blocks first
+        assert not node0.waitForLibToAdvance(timeout=5), "LIB should not advance on node0 after centerNode was shutdown"
+    if node1.waitForLibToAdvance(timeout=5):
+        assert not node1.waitForLibToAdvance(timeout=5), "LIB should not advance on node1 after centerNode was shutdown"
+
+    Print("Restart centerNode")
+    centerNode.relaunch()
+
+    Print("Verify LIB advances after restart")
+    assert node0.waitForLibToAdvance(), "node0 did not advance LIB"
+    assert node1.waitForLibToAdvance(), "node1 did not advance LIB"
+    assert producercNode.waitForLibToAdvance(), "producercNode did not advance LIB"
+
+    ####################### test 3 ######################
+
+    Print("Shutdown producercNode")
+    producercNode.kill(signal.SIGTERM)
+    assert not producercNode.verifyAlive(), "producercNode did not shutdown"
+
+    # disable production-pause-vote-timeout
+    Print("Relaunch producercNode with --production-pause-vote-timeout-ms 0")
+    addSwapFlags={"--production-pause-vote-timeout-ms": "0"}
+    producercNode.relaunch(chainArg="--enable-stale-production", addSwapFlags=addSwapFlags)
+
+    Print("Shutdown finalizercNode")
+    finalizercNode.kill(signal.SIGTERM)
+    assert not finalizercNode.verifyAlive(), "finalizercNode did not shutdown"
+
+    # Verify producercNode still producing
+    assert producercNode.waitForHeadToAdvance(), "producercNode (--production-pause-vote-timeout-ms 0) paused after finalizercNode was shutdown"
```

Comment:
Earlier you said:

```
        # Do not use waitForHeadToAdvance() to check for pausing, as producercNode
        # still receive blocks from node0 and node1 and can make head advance
```

So the check here that head is advancing does not tell us anything about whether `producecNode` paused production or not.

You still need to use the endpoint to check that it has not paused and repeat that for some number of seconds (at least more than 6) to be convinced that it will not pause.